presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -35,7 +35,18 @@
     private static final Capture<AggregationNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
-            .with(source().matching(aggregation().capturedAs(CHILD)));
+            .with(source().matching(aggregation().capturedAs(CHILD)
```

Comment:
I believe the '1 property per row' rule should be used here over the '1 chained method per row' rule.